Added cloneTo method to support trajectory point copying.

shaun-edwards opened the pull request:
Addresses issue #27.  Refactored/expanded test suite.

shaun-edwards committed:
Added cloneTo method to support trajectory point copying.  Refactored/expanded test suite

shaun-edwards commented:
@dpsolomon, a quick one to check.

dpsolomon commented on descartes_core/test/trajectory_pt.cpp:
```diff
@@ -61,23 +74,31 @@ typedef Types<CartTrajectoryPt, JointTrajectoryPt> Implementations;
+  //Copied objects should always be the same
+  //TODO: The equal comparison fails for some unknown reason.  In the debugger
+  //the values are equal, so this may be a problem with eigen.
+  //EXPECT_EQ(this->lhs_, this->lhs_copy_);
```
I think you are comparing pointers, which should return false in this case. Did you try 
`EXPECT_EQ(*(this->lhs_), *(this->lhs_copy_));`?

dpsolomon commented:
+1 @shaun-edwards . If you want to fix the test first let me know, if you don't care about it too much I'll just merge it.

shaun-edwards committed:
Fixed comparison check (not compare equal value, but this isn't implemented yet).

shaun-edwards commented:
Pointers...well that's embarrassing.  I fixed the tests to reflect what I really wanted...and then commented them out because that functionality is missing.  I added issue #32 to address this in the future.  @jrgnicho needs the clone functionality ASAP, so I'd prefer to accept the PR as is.  @dpsolomon, can you merge?

dpsolomon commented:
Next time I'll send a private note ;)
